mantidproject/mslice
# Scale and Rebose workspace operations

Implements a `Scale` and `Rebose` (temperature scaling) button.

**This PR should be merged *after* PR 471 or the script generation will not work correctly**

**To test:**



Start MSlice and load these data files: [bose_data.zip](https://github.com/mantidproject/mslice/files/3354975/bose_data.zip). Select the 300K dataset and click the arrow next to the `Compose` button and then click the `Bose` button. A dialog should pop up which asks for the current sample temperature (`300`) and the target temperature (`5`). Rescale the dataset from 300K to 5K. Plot the rescaled data and compare it to the actual 5K data, by doing a cut along `DeltaE` from `5` to `80` meV integrating over `|Q|` from `5` to `10` reciprocal Angstroms. You should see that the rescaled curve should match the actual 5K data better at low energies (10-30meV) than the original data. This is because 300K is approximately 30meV and at high temperatures you get additional scattering from thermally populated modes at these energies, whilst the high energy scattering is mostly the same for the two temperatures.

(If this is done after PR 471 is merged, you can also click on `Generate Script` in the `File` menu and then inspect and run the script to reproduce the plot).

Select one of the workspace and click on the arrow button next to the `Bose` button and select `Scale`. A dialog should pop up asking for a scale factor - put in a number. Plot the scaled data and check that it has been scaled correctly.


Fixes #463
